### DISCUSSION TITLE
Hardware audio inputs to Hubs (via Firefox/Chrome on MacOS)

### DISCUSSION THREAD

**Discussion by m2u3** [mozilla/hubs#2842] [General Questions & Discussion] [answered]
Hello...

I'm trying to set up live a AV stream (built-in cam, external audio coming in via soundcard) to hubs and have tried both Chrome and Firefox (MacOS 10.14). I can pipe audio through into Chrome without problem by selecting my soundcard as a source when first entering a room and specifying the Microphone – but the audio has audible compression artefacts. 

Using Firefox, I get no sound (even though the soundcard shows up as an input choice) unless I use a virtual patchbay such as LoopBack – in which case the audio sounds much better than via Chrome. 

I don't see any controls in the browsers beyond granting mic permissions. 

Anyone out there have experience of piping audio from an external soundcard via different browsers? Any tips appreciated – thanks!

Mukul

(soundcard is MIO 2882)

**Comment by brianpeiris** [comment, marked as answer]
Hi @m2u3. Using the voice channel to stream arbitrary audio might not be the best option, since I believe that channel is tuned specifically to carry human voice audio. You might be better off streaming from something like [OBS](https://obsproject.com/) via Twitch or YouTube (youtube is a secondary choice since it can be flakey). You could also stream via another third-party service that you pay for, if reliability is key. Essentially, Hubs will accept any video stream that is available publicly on the web.

**Reply by m2u3** [reply]
Hi Brian –
Thanks a lot for your reply!
I'm curious about the channel being tuned for voice, since wouldn't it also be used for sharing say a video file playing back on the desktop?
I've now realised that it's better to set up the streaming windows in a Spoke scene anyway (if nothing else, so that visitors to the room can't move them around accidentally) and I'm using Twitch now.  I'd rather use OBS but had problems with it recognising multichannel sound cards (this may be a MacOS specific issue), so I ended up using Twitch Studio Beta. Ideally I'd look for an open (or at least non-Amazon) solution, but for now this will do. One issue is that  Twitch isn't available to our colleagues in China (but then sadly nor is Hubs, without a VPN).
